### PR TITLE
[pilot] do not advertise the `skip_waiting_for_build_processing` option when it is already set

### PR DESCRIPTION
**Pull request by robinkunde**
### Checklist

- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.
- [x] I've added or updated relevant unit tests.

### Motivation and Context

Reduces superfluous output when uploading builds

### CODE REVIEW

**Review by lacostej — CHANGES_REQUESTED** [review]
Hello,

Thanks a lot for the contribution.

Can you please rebase your PR on top of master, to fix the build checks. Also consider the comment.

Thanks!

File: pilot/lib/pilot/build_manager.rb [hunk 1]
```diff
@@ -72,8 +72,10 @@ def upload(options)
       # Calling login again here is needed if login was not called during 'start'
```
[the remark this answers is not in the document]
Comment (robinkunde) [reply pilot/lib/pilot/build_manager.rb:72]: Will do

[the remark this answers is not in the document]
Comment (robinkunde) [reply pilot/lib/pilot/build_manager.rb:73]: @lacostej Please take another look when you get a chance

**Review by lacostej — APPROVED** [review]
Thanks a lot for the effort :) LGTM

File: pilot/lib/pilot/build_manager.rb [hunk 2]
```diff
@@ -65,15 +65,18 @@ def upload(options)
...
       end
 
       # Calling login again here is needed if login was not called during 'start'
       login unless should_login_in_start
 
-      UI.message("If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option")
-      UI.message("Note that if `skip_waiting_for_build_processing` is used but a `changelog` is supplied, this process will wait for the build to appear on AppStoreConnect, update the changelog and then skip the remaining of the processing steps.")
+      if config[:skip_waiting_for_build_processing].nil?
+        UI.message("If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option")
+        UI.message("Note that if `skip_waiting_for_build_processing` is used but a `changelog` is supplied, this process will wait for the build to appear on AppStoreConnect, update the changelog and then skip the remaining of the processing steps.")
```
Comment (rogerluan) [inline pilot/lib/pilot/build_manager.rb:78]: I know this wasn't written in this PR but I think we can improve it :)

```suggestion
        UI.message("Note that if `skip_waiting_for_build_processing` is used but a `changelog` is supplied, this process will wait for the build to appear on App Store Connect, update the changelog and then skip the remaining of the processing steps.")
```

File: pilot/lib/pilot/build_manager.rb [hunk 3]
```diff
@@ -65,15 +65,18 @@ def upload(options)
           UI.important("`skip_waiting_for_build_processing` used and no `changelog` supplied - skipping waiting for build processing")
           return
         else
+          UI.important("`skip_waiting_for_build_processing` used and `changelog` supplied - will wait until build appears on AppStoreConnect, update the changelog and then skip the rest of the remaining of the processing steps.")
```
Comment (rogerluan) [inline pilot/lib/pilot/build_manager.rb:68]:
```suggestion
          UI.important("`skip_waiting_for_build_processing` used and `changelog` supplied - will wait until build appears on App Store Connect, update the changelog and then skip the rest of the remaining of the processing steps.")
```

File: pilot/spec/build_manager_spec.rb [hunk 4]
```diff
@@ -599,6 +599,53 @@
...
+
+        fake_build = double
+        expect(fake_build_manager).to receive(:wait_for_build_processing_to_be_complete).and_return(fake_build)
+
+        expect(fake_build_manager).to receive(:distribute).with(upload_options, build: fake_build)
+      end
+
+      it "does not advertise `skip_waiting_for_build_processing` if the option is set" do
+        expect(FastlaneCore::UI).to_not(receive(:message).with("If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option"))
+        expect(FastlaneCore::UI).to_not(receive(:message).with("Note that if `skip_waiting_for_build_processing` is used but a `changelog` is supplied, this process will wait for the build to appear on AppStoreConnect, update the changelog and then skip the remaining of the processing steps."))
```
Comment (rogerluan) [inline pilot/spec/build_manager_spec.rb:637]:
```suggestion
        expect(FastlaneCore::UI).to_not(receive(:message).with("Note that if `skip_waiting_for_build_processing` is used but a `changelog` is supplied, this process will wait for the build to appear on App Store Connect, update the changelog and then skip the remaining of the processing steps."))
```

File: pilot/spec/build_manager_spec.rb [hunk 5]
```diff
@@ -599,6 +599,53 @@
...
+
+        fake_build_manager.upload(upload_options)
+      end
+
+      it "shows notice when using `skip_waiting_for_build_processing` and changelog together" do
+        expect(FastlaneCore::UI).to(receive(:important).with("`skip_waiting_for_build_processing` used and `changelog` supplied - will wait until build appears on AppStoreConnect, update the changelog and then skip the rest of the remaining of the processing steps."))
```
Comment (rogerluan) [inline pilot/spec/build_manager_spec.rb:643]:
```suggestion
        expect(FastlaneCore::UI).to(receive(:important).with("`skip_waiting_for_build_processing` used and `changelog` supplied - will wait until build appears on App Store Connect, update the changelog and then skip the rest of the remaining of the processing steps."))
```

**Review by rogerluan — APPROVED** [review]
Thank you for your contribution @robinkunde 🤗 this LGTM!

**Comment by lacostej** [comment]
Thanks @robinkunde :)

**Comment by rogerluan** [comment]
Thanks for fixing the issue I accidentally caused @lacostej ❤️

**Pull request merged**